### ISSUE TITLE
`convnext` implementations fail on CUDA GPUs

### ISSUE THREAD

**Issue by djpetti** [pytorch/vision#7124]
### 🐛 Describe the bug

I'm not entirely sure if this is reproduceable, or just a quirk of my environment. For what it's worth, I've tried it on several GPUs, and gotten the same result. The problem is with the built-in `convnext` implementation, and can be triggered like so:

```python
import torch
from torchvision.models import convnext_tiny

images = torch.normal(0.0, 1.0, size=(2, 3, 100, 100), dtype=torch.float32).to("cuda")
model = convnext_tiny().to("cuda")
model(images)
```

Running on the CPU, this outputs the class probabilities like you would expect. However, on the GPU, it crashes:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torchvision/models/convnext.py", line 176, in forward
    return self._forward_impl(x)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torchvision/models/convnext.py", line 170, in _forward_impl
    x = self.features(x)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/container.py", line 204, in forward
    input = module(input)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/container.py", line 204, in forward
    input = module(input)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torchvision/models/convnext.py", line 63, in forward
    result = self.layer_scale * self.block(input)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/container.py", line 204, in forward
    input = module(input)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/blue/cli2/daniel.petti/ssl/.venv/lib/python3.8/site-packages/torch/nn/modules/linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: CUDA error: CUBLAS_STATUS_INVALID_VALUE when calling `cublasSgemm( handle, opa, opb, m, n, k, &alpha, a, lda, b, ldb, &beta, c, ldc)`
```

### Versions

```
Collecting environment information...
PyTorch version: 1.13.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Red Hat Enterprise Linux Server release 7.9 (Maipo) (x86_64)
GCC version: (crosstool-NG 1.24.0.133_b0863d8_dirty) 9.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.8.5 (default, Sep  4 2020, 07:30:14)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-3.10.0-1160.81.1.el7.x86_64-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.4.152
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 Ti
GPU 1: NVIDIA GeForce RTX 2080 Ti
GPU 2: NVIDIA GeForce RTX 2080 Ti
GPU 3: NVIDIA GeForce RTX 2080 Ti
GPU 4: NVIDIA GeForce RTX 2080 Ti
GPU 5: NVIDIA GeForce RTX 2080 Ti
GPU 6: NVIDIA GeForce RTX 2080 Ti
GPU 7: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 470.161.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.0
[pip3] torch==1.13.1
[pip3] torchvision==0.14.1
[conda] Could not collect
```

For reference, `torch.backends.cudnn.version()` reports `8500`, even though the script didn't get the CuDNN version for some reason.

**Comment by NicolasHug** [comment]
Hi @djpetti ,

Unfortunately I'm not able to reproduce the issue:

```py
[ins] In [1]: import torch
         ...: from torchvision.models import convnext_tiny
         ...: 
         ...: images = torch.normal(0.0, 1.0, size=(2, 3, 100, 100), dtype=torch.float32).to("cuda")
         ...: model = convnext_tiny().to("cuda")
         ...: model(images)
Out[1]: 
tensor([[ 0.3778, -0.3301,  0.5875,  ...,  0.3186,  0.7818,  0.6760],
        [ 0.2024,  0.0881, -0.0515,  ..., -0.1618,  1.3115, -1.2001]],
       device='cuda:0', grad_fn=<AddmmBackward0>)
```

(I ran this on a A100)

Googling the error message suggests it might be memory-related. But I tried a `(5, 3, 1000, 1000)` image and I'm getting a  proper `OutOfMemoryError`, i.e. not the same error message. So... I don't know (sorry).

Out of curiosity, what happens if you pass a `(1, 3, 32, 32)` image?

**Comment by djpetti** [comment]
Yeah, that was my first thought too, but I found that I got the same error with inputs of any batch size. Later, I figured out that this error is not actually limited to `convnext`. It appeared in all sorts of other (seemingly random) situations.

The system that I was testing this on actually has a system-level CUDA 11.4 installation. At the time, I didn't think much of it, since Pytorch bundles its own CUDA. However, I figured out that if I uninstalled the system CUDA, then the Pytorch issues went away! So I think there must have been some kind of weird CUDA version conflict between the bundled and system CUDAs.

Anyway, I'm closing this issue, because it's an environment problem and not a bug in Pytorch.
